pkg/airgap/update.go

```diff
@@ -315,26 +316,66 @@ func canInstall(beforeKotsKinds *kotsutil.KotsKinds, afterKotsKinds *kotsutil.Ko
 	return nil
 }
 
-func GetMissingRequiredVersions(appID string, airgap *kotsv1beta1.Airgap) ([]string, error) {
-	appVersions, err := store.GetStore().GetAppVersions(appID)
+func GetMissingRequiredVersions(app *apptypes.App, airgap *kotsv1beta1.Airgap) ([]string, error) {
+	appVersions, err := store.GetStore().GetAppVersions(app.ID)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get downstream versions")
 	}
 
-	installedVersions := make(map[string]bool)
-	for _, appVersion := range appVersions {
-		if appVersion.ChannelID != airgap.Spec.ChannelID {
-			continue
-		}
-		installedVersions[appVersion.UpdateCursor] = true
+	license, err := kotsutil.LoadLicenseFromBytes([]byte(app.License))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load license")
 	}
 
+	return getMissingRequiredVersions(airgap, license, appVersions)
+}
+
+func getMissingRequiredVersions(airgap *kotsv1beta1.Airgap, license *kotsv1beta1.License, installedVersions []*types.AppVersion) ([]string, error) {
 	missingVersions := make([]string, 0)
+	if len(installedVersions) == 0 {
+		return missingVersions, nil
+	}
+
 	for _, requiredRelease := range airgap.Spec.RequiredReleases {
-		if installedVersions[requiredRelease.UpdateCursor] {
+		laterReleaseInstalled := false
+		for _, appVersion := range installedVersions {
+			if license.Spec.IsSemverRequired {
+				// semvers can be compared across channels
+				if appVersion.Semver == nil {
```

Comment:
semver and non-semver can live together. we probably shouldn't fail here, but rather just continue?